src/System.Management.Automation/security/wldpNativeMethods.cs

```diff
@@ -696,6 +696,7 @@ internal static class WldpNativeMethods
             [DefaultDllImportSearchPathsAttribute(DllImportSearchPath.System32)]
             [DllImportAttribute("wldp.dll", EntryPoint = "WldpCanExecuteFile")]
             internal static extern int WldpCanExecuteFile(
+                [MarshalAs(UnmanagedType.LPStruct)]
```

Comment:
Should we add a comment here that this is needed because the first argument is a `REFGUID`?

Reply:
The docs for `LPStruct` DO indicate that this should ONLY be used for REFGUID, but someone not familiar with that may not understand the connection between this type and GUID.

Reply:
I have made a suggestion to use the `in` modifier instead.

Reply:
To allow anyone not familiar with the subject to understand why the attribute is there, you'd have to include a treatise on calling conventions and how C weirdly came up with the idea of passing structs on the stack. Anything less than that just ends up as a string of the same words you are trying to explain.

Link to https://learn.microsoft.com/en-us/dotnet/standard/native-interop/best-practices#guids ? That is what you are referring to above, I believe. That sentence doesn't say "the only type in the whole world that LPStruct should ever be used with is Guid", but "of the various ways of declaring a Guid argument, LPStruct should be used only with the `ref` kind". (If I find the time, I'll try and improve the Guid section of that page. I find it very confusing.)

Reply:
> I have made a suggestion to use the `in` modifier instead.

Doesn't `in` just mean the argument is effectively `const` on the receiving end? I admit I have never used it, but it does not look to me like it also means to pass the thing as a pointer.

Reply:
> Doesn't `in` just mean the argument is effectively `const` on the receiving end? I admit I have never used it, but it does not look to me like it also means to pass the thing as a pointer.

Oh, I take that back.